File: Languages/Ruby/Day009.rb
# Languages - Ruby - Day 9
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a program that uses a loop to print the numbers from 1 to 10 and calculates their sum.

def calculate_sum_in_range(start_range, end_range)
  sum = 0
  for i in start_range..end_range
    puts i
    sum += i
  end
  puts "The sum of the numbers from #{start_range} to #{end_range} is: #{sum}"
end

calculate_sum_in_range(1, 10)

File: Languages/TypeScript/Day009.ts
// Languages - TypeScript - Day 9
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a program that uses a loop to print the numbers from 1 to 10 and calculates their sum.

function calculateSumInRanges(start: number, end: number): void {
    let sumOfNumbers: number = 0;
    for (let i = start; i <= end; i++) {
        console.log(i);
        sumOfNumbers += i;
    }
    console.log(`The sum of the numbers from ${start} to ${end} is ${sumOfNumbers}`);
}

calculateSumInRanges(1, 10);
